File: primo.py
import math

import psycopg2
from tabulate import tabulate
def  post(x,y):
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "jblue5queso",
            dbname = "tarea1"
            )
    except:
        print("Sin Conexion Exitosa\n")
            
    cursor = conexion.cursor()
    cursor.execute("INSERT INTO primo(num, Es) VALUES (%s, %s);", (x, y))
    conexion.commit()
    cursor.close()
    conexion.close()

def ret():
    try:
        conexion = psycopg2.connect(
            host = "localhost",
            port = "5432",
            user = "postgres",
            password = "jblue5queso",
            dbname = "tarea1"
            )
    except:
        print("Sin Conexion Exitosa\n")
    
    cursor = conexion.cursor()
    cursor.execute("SELECT * from primo;")
    print(tabulate(cursor, headers=["ide", "num", "Es"], tablefmt="orgtbl", numalign ="center"))


while True:
    try:
        print("Elija una opcion: 1. Saber si un numero es primo o compuesto,   2. Historial,  3.Salir: ")
        x=int(input("Opcion a elegir: \n"))
        if x>3 or x<=0:
            print("Numero erroneo, vuelva a ingresar: \n")
        if x==1:
            num=int(input("Ingrese numero a comprobar: "))
            primo=True
            Es=""
            for i in range(2,num):
                if num%i==0:
                    print("Es un numero compuesto")
                    Es="Es compuesto"
                    primo=False
                    break
            if primo:
                print("Es un numero primo")
                Es="Es un numero primo"
            post(num, Es)
        elif x==2:
            ret()
            break
        elif x==3:
            break

    except Exception as es:
        print("\n Ingreso un caracter y no un numero, vuelva a intentar:\n")
        print(repr(es))

exit(0)

File: primo.m
pkg load database;
clc; 
clear all; 
close all;

retry= true;
while retry
  try
    fprintf("Elija una opcion: 1. Saber si un numero es primo o compuesto,   2. Historial)");
    i=(input ("\n Opcion a elegir: "));
    if i>3||i<0
      fprintf("\n Numero erroneo, vuelva a ingresar: ");
    endif
      if i==1
        num=input("Ingrese el numero a verificar: ")
        n=0;
        Es='';
        for x=1:num
        r=rem(num,x);
        if r==0;
          n=n+1;
        endif
        endfor
        if n==2
          fprintf("Este numero es primo \n");
          Es='Es un numero primo';
        else
            fprintf("Este numero es compuesto \n");
            Es='Es compuesto';
        endif
        %Conectar con PGADMIN
        conn = pq_connect(setdbopts('dbname','tarea1','host','localhost',
        'port','5432','user','postgres','password','jblue5queso'));
        params=cell(1,2);
        params{1,1}=num;
        params{1,2}=Es;
        N=pq_exec_params(conn, "insert into primo(num, Es) values($1,$2);",params); %insertar datos en la tabla
        %Texto
        save('primo.txt','num', 'Es','-ascii');

    endif
    if i==2
       conn = pq_connect(setdbopts('dbname','tarea1','host','localhost',
       'port','5432','user','postgres','password','jblue5queso'));
       N=pq_exec_params(conn, "Select * from primo;");%historial de la tabla
       disp(N);
       retry=false;
    endif
   catch
      fprintf("Solo se permiten valores numericos, vuelva a selecionar \n");
      msg = lasterror.message;
      fprintf(msg);
    end_try_catch
 endwhile